code/modules/reagents/chemistry/holder.dm

```diff
@@ -299,10 +300,7 @@
 			if(R.addiction_stage < 5)
 				if(prob(5))
 					R.addiction_stage++
-			if(M.reagents.has_reagent(R.id))
-				R.last_addiction_dose = world.timeofday
-				R.addiction_stage = 1
-			else
+			if(world.timeofday > (R.last_addiction_dose + ADDICTION_WITHDRAWAL_DELAY)) //time check so addiction act doesn't play over and over. Allows incremental dosages to work.
```

Comment:
I'm pretty sure that `if(world.timeofday > R.last_addiction_dose)` Would be enough.

You can throw some debug `to_chat(world, "blehhh")` in to test this with smoking cigs or something similar just to be sure that what's below this `if` statement isn't running---you can even see how much time has past by throwing in a `to_chat(world, "[world.timeofday]")` on `reagent/on_mob_life` and right about here to see how much time has passed; I'd venture to guess the game won't even register a time change, as this only has granularity on the order of 1/10th of a second, and reagent processing is not really knowing to induce lag (unless something horrendous went wrong).

Of the edge cases of there being lag, this is so inconsequential that `addiction_act_stage1` running once in a blue moon, only to return to proper behavior the next time this runs....I'm not worried.

Reply:
yeah that works